src/SFML/Audio/Music.cpp

```diff
@@ -109,36 +110,171 @@ Time Music::getDuration() const
 
 
 ////////////////////////////////////////////////////////////
+Uint64 Music::getSampleCount() const
+{
+    return m_file.getSampleCount();
+}
+
+
+////////////////////////////////////////////////////////////
+Music::TimeSpan Music::getLoopPoints() const
+{
+    TimeSpan out(Time::Zero, Time::Zero);
+
+    // Make sure we don't divide by 0
+    if (getSampleRate() != 0 && getChannelCount() != 0)
+    {
+        out.offset = samplesToTime(m_loopSpan.offset);
+        out.length = samplesToTime(m_loopSpan.length);
+    }
+
+    return out;
+}
+
+////////////////////////////////////////////////////////////
+void Music::setLoopPoints(TimeSpan timePts)
+{
+    SampleSpan samPts = SampleSpan(timeToSamples(timePts.offset), timeToSamples(timePts.length));
+
+    // Check our state. This averts a divide-by-zero. GetChannelCount() is cheap enough to use often
+    if (getChannelCount() == 0 || m_file.getSampleCount() == 0)
+    {
+        sf::err() << "Music is not in a valid state to assign Loop Points." << std::endl;
+        return;
+    }
+
+    // Round up to the next even sample if needed
+    samPts.offset += (getChannelCount() - 1);
+    samPts.offset -= (samPts.offset % getChannelCount());
+    samPts.length += (getChannelCount() - 1);
+    samPts.length -= (samPts.length % getChannelCount());
+
+    // Validate
+    if (samPts.offset >= m_file.getSampleCount())
+    {
+        sf::err() << "LoopPoints offset val must be in range [0, Duration)." << std::endl;
+        return;
+    }
+    if (samPts.length == 0)
+    {
+        sf::err() << "LoopPoints length val must be nonzero." << std::endl;
+        return;
+    }
+
+    // Clamp End Point
+    samPts.length = std::min(samPts.length, m_file.getSampleCount() - samPts.offset);
+
+    // If this change has no effect, we can return without touching anything
+    if (samPts.offset == m_loopSpan.offset && samPts.length == m_loopSpan.length)
+        return;
+
+    // When we apply this change, we need to "reset" this instance and its buffer
+
+    // Get old playing status and position
+    Status oldStatus = getStatus();
+    Time oldPos = getPlayingOffset();
+
+    // Unload
+    stop();
+
+    // Set
+    m_loopSpan = samPts;
+
+    // Restore
+    if (oldPos != Time::Zero)
+        setPlayingOffset(oldPos);
+
+    // Resume
+    if (oldStatus == Playing)
+        play();
+}
+
+
+////////////////////////////////////////////////////////////
 bool Music::onGetData(SoundStream::Chunk& data)
 {
     Lock lock(m_mutex);
 
+    std::size_t toFill = m_samples.size();
+    Uint64 curOffset = m_file.getSampleOffset();
+    Uint64 loopEnd = m_loopSpan.offset + m_loopSpan.length;
+
+    // If the loop end is enabled and imminent, request less data.
+    // This will trip an "onLoop()" call from the underlying SoundStream,
+    // and we can then take action.
+    if (getLoop() && m_loopSpan.length != 0 && curOffset <= loopEnd && curOffset + toFill > loopEnd)
```

Comment:
Every sub-expression of the condition should be enclosed in ( ). Same applies to other parts of the code.